### ISSUE TITLE
PHP Deprecated:  Hook woocommerce_rest_api_option_permissions is deprecated since version 6.3.0

### ISSUE THREAD

**Issue by ldr2273** [Automattic/woocommerce-payments#7864]
### Describe the bug

After the update to WooCommerce Payments 6.9.1, the following notice is triggered:

```
PHP Deprecated:  Hook woocommerce_rest_api_option_permissions is deprecated since version 6.3.0 with no alternative available. in /wordpress/core/6.4.2/wp-includes/functions.php on line 6031
```

### To Reproduce

1. Enable WP_DEBUG_LOG
2. Update WooCommerce Payments to version 6.9.1.
3. Debug logs reveal the notice

### Actual behavior
The notice is triggered after the update.

Most likely because of this: https://github.com/Automattic/woocommerce-payments/blob/6.9.0/includes/class-wc-payments.php#L581

Which has been deprecated by WooCommerce Admin here: https://github.com/woocommerce/woocommerce/blob/8.3.1/plugins/woocommerce/src/Admin/API/Options.php#L149

### Screenshots
Replicated:

<img width="1422" alt="Screenshot 2023-12-08 at 7 30 08 AM" src="https://github.com/Automattic/woocommerce-payments/assets/20307519/ac9d7c64-41ea-4e5b-9fa5-3bac616f2d03">

### Expected behavior
We would migrate to the new version of that hook and the notice would not be triggered.

### Desktop (please complete the following information):

* OS: Sonoma
* Browser: Chrome

**Comment by ldr2273** [comment]
Created a report here as well, since core needs to find a replacement for the deprecated API: https://github.com/woocommerce/woocommerce/issues/41960

**Comment by RadoslavGeorgiev** [comment]
Blocked, pending a response in Core: paJDYF-bfd-p2#comment-22185

**Comment by glagonikas** [comment]
I'm surprised this is still not resolved, we reported over a month ago and it's quite a common issue.

We've ended up with a hardcode as follows

in the file - wp-content\plugins\woocommerce\src\Admin\API\Options.php at line 149 change:

`return apply_filters_deprecated( 'woocommerce_rest_api_option_permissions', array( $permissions, $request ), '6.3.0' );`

to

`return apply_filters( 'woocommerce_rest_api_option_permissions', array( $permissions, $request ) );`

‌and for the other warning that will come up, comment out line 115:

`wc_deprecated_function( 'Automattic\WooCommerce\Admin\API\Options::' . ( $is_update ? 'update_options' : 'get_options' ), '6.3' );`

**Comment by shaunkuschel** [comment]
Received another report of this on 7817483-zen

**Comment by Nyota** [comment]
Is this causing 500 Internal Error messages for anyone?

**Comment by shameemreza** [comment]
Another report here: 8284246-zen
Thread: https://wordpress.org/support/topic/woocommerce_rest_api_option_permissions/

**Comment by stephondoestech** [comment]
My client is also having this issue. When they try to upload images in the app we get the following: 

[12-Jul-2024 00:51:48 UTC] PHP Deprecated:  Hook woocommerce_rest_api_option_permissions is deprecated since version 6.3.0 with no alternative available. in /home/test/public_html/wp-includes/functions.php on line 6078

It's been over 7 months since this issue was first reported and it's not even assigned for work.

**Comment by OmarFPG** [comment]
8460719-zen

**Comment by OmarFPG** [comment]
> Blocked, pending a response in Core: paJDYF-bfd-p2#comment-22185

Do we have any updates on this? @RadoslavGeorgiev

**Comment by giuseal** [comment]
I am also having problems caused by this thing.

**Comment by bomsn** [comment]
having the same problem here, and a lot of other issues.

**Comment by EjayhanFernandes** [comment]
8526031-zen

**Comment by silviofei** [comment]
Tons of debug.log, please fix it, too spammy

**Comment by tonydream888** [comment]
It flood the debug.log and make it very difficult to read and find other problems!

**Comment by RiaanKnoetze** [comment]
Related: https://wordpress.org/support/topic/cannot-load-woopayments-overview-page/#post-17999411

**Comment by RadoslavGeorgiev** [comment]
This issue is probably fixed already, as those notices don't seem to appear anymore. Let's use the opportunity to confirm that, and close this issue.

**Comment by giuseal** [comment]
I still have this issue.

**Comment by htdat** [comment]
Thanks for the update @giuseal! 

> This issue is probably fixed already, as those notices don't seem to appear anymore.

@RadoslavGeorgiev - it's a pity that the issue is still there and it can not be resolved until we have a long-term solution paJDYF-bfd-p2#comment-22405 or [replacement](https://github.com/woocommerce/woocommerce/issues/41960) in Woo core. 

Try to see if we can have a solution soon p1741839271739529-slack-C0E1AV8T0

**Comment by RadoslavGeorgiev** [comment]
Yes, after reading @giuseal's comment above, I realized that I have intentionally commented out a couple of lines in my local WP instance in order to silence this message until it gets fixed. Thanks for moving it forward @htdat, I'll be monitoring that thread.

**Comment by giuseal** [comment]
Is this thread closed? Solved? Because for me, the issue, the reason why I was in this thread, persists.

**Comment by RadoslavGeorgiev** [comment]
@giuseal https://github.com/Automattic/woocommerce-payments/pull/10636 was added to WooPayments 9.2.0, released on April 9. We are not using the options API, which triggered the warning, at all. Have you updated to 9.2.0? If so, are you sure that the log originates from WooPayments?

**Comment by giuseal** [comment]
@RadoslavGeorgiev Thanks for your response. I was following this thread because the people in the WooCommerce forum suggested it to me. My website during the day time to time cannot be loaded for a couple of minutes. The support told me that "the issue lies within the WooPayments plugin itself and our development team is currently aware of it and working towards a solution". I have version 9.2.0.

**Comment by RadoslavGeorgiev** [comment]
@giuseal this issue is regarding a deprecated hook, which is no longer used in WooPayments 9.2.0, a topic that must have been addressed. In any case, on production sites, this deprecation notice would only be added to logs, and should not prevent a website from loading. Are you sure that this is the 9.2.0 feature that you have been pointed to?

**Comment by giuseal** [comment]
Thank you @RadoslavGeorgiev 
Yes I am sure. I’m checking it right now.
